client/components/CreateProfileAddInfo.js

```diff
@@ -0,0 +1,189 @@
+import React, { Component } from 'react';
+import { View, ScrollView, StyleSheet, Keyboard, Dimensions, TouchableOpacity, Text } from 'react-native';
+import { Formik } from 'formik';
+import { connect } from 'react-redux';
+import { storeMyId }  from '../redux/actions/actions';
+import { TextField } from 'react-native-material-textfield';
+
+import { withNavigation } from 'react-navigation';
+import { populateNewProfile } from '../utils/profile-obj-compress';
+
+const { width } = Dimensions.get('window');
+// const minDate= "1900-05-01";
+// const maxDate = "2016-06-01";
+const minDate= "01 Jan 1900";
+const maxDate = "01 Jan 2001";
+const BASE_URL = "http://192.168.1.149:3000";
+
+class FormCreateProfileAddInfo extends Component {
+  
+  static navigationOptions = {
+    title: 'Add Info',
+  };
+
+  createMyProfileInDB = async (newProfile) => {
+    let payload = JSON.stringify(newProfile);
+
+    await fetch(`${BASE_URL}/user`, {
+      method: "POST",
+      headers: { 'Content-Type': 'application/json' },
+      body: payload
+    })
+      .then(rawData => rawData.text())
+      .then((myProfileID) => {
+
+        myProfileIdCleaned = myProfileID.replace(/"/g, '');
+        this.props.storeMyId(myProfileIdCleaned);
+      })
+      .catch( (err) => console.log('ERROR IN CREATING THE USER',err));
+      // this.props.navigation.navigate('Dashboard');
+  };
+
+  render() {
+    return (
+      <View style={styles.container}>
+
+      <Formik 
+            initialValues={{ company: '', nationality:'', country: '',  place: '', linkedIn: '', github: '', cv: '', website: '' }} 
+
+            onSubmit={ values => {
+
+              const newProfileObject = this.props.navigation.getParam('newProfile', {});
+              const { company, nationality, country,  place, linkedIn, github, cv, website } = values;
+
+                Keyboard.dismiss();
+                let completedProfile = populateNewProfile( newProfileObject, company, nationality, country,  place, linkedIn, github, cv, website);
+                
+                delete completedProfile._id;
+                
+                ( async () => {
+                await this.createMyProfileInDB(completedProfile);
+                console.log('COMPLETED PROFILE', completedProfile);
+                  this.props.navigation.navigate('Dashboard');
+                })();
+              }
+            }>
+          {({ handleChange, handleSubmit, values, handleBlur }) => (
+
+            <ScrollView 
+              contentContainerStyle={styles.wrapper}
+              showsVerticalScrollIndicator={false}
+              >
+            <TextField
```

Comment:
Same as above I think making one big state object and mapping over it will help with legibility and cut down on repetative code